### PR TITLE
Fix `StateVersionResources` field types

### PR DESCRIPTION
**Pull request by derekmckinnon**
## Description

When calling the `StateVersions.ReadCurrent` function, I noticed that the return `StateVersion` struct's `Resources` slice was always empty, even though `ResourcesProcessed` was `true` and my TFE state version definitely has resources. I used `curl` against the same endpoint and received the resources as expected, which led me to believe there was an unmarshalling error going on.

After debugging for a while, I found that the underlying `jsonapi` dependency was silently discarding the `StateVersionResources` with an error message of: `"The struct field was not of a known number type"`. I went to the code and realized that the `StateVersionResources` `Count` and `Type` had their data types swapped.

Changing `Count` to `int` and `Type` to `string` resolved the issue. I was able to testing using my forked version and swapping that in to my app. I had a look at the existing test suites in this project and I hesitate to change them because none of them seem to check for `StateVersionResources` at this time.

## Testing plan

1. Use the SDK to call any of the `StateVersions.Read*` functions on a workspace that has resources
2. Output the values of `ResourcesProcessed` and `StateVersionResources`

You will see that `ResourcesProcessed == true` and `len(StateVersionResources) == 0` .

### CODE REVIEW

**Review by laurenolivia — APPROVED** [review]
Good catch

**Comment by derekmckinnon** [comment]
Hmm, seems like some of the tests are failing due to authentication issues with the CI tasks?

**Comment by laurenolivia** [comment]
@derekmckinnon I will resolve this today

**Comment by laurenolivia** [comment]
@derekmckinnon I had to open a new PR #739 and cherry-pick your commit so I could run CI tests on an internal token. You've been linked to this new PR, and tests are passing. Will merge this once someone on my team approves.

**Comment by laurenolivia** [comment]
Closing this PR, now that #739 has been merged.

**Pull request closed without merging**
